brainiak/fcma/mvpa_voxelselector.py

```diff
@@ -0,0 +1,195 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Full Correlation Matrix Analysis (FCMA)
+
+Activity-based voxel selction
+"""
+
+# Authors: Yida Wang
+# (Intel Labs), 2017
+
+import numpy as np
+from sklearn import model_selection
+from scipy.stats.mstats import zscore
+import logging
+from mpi4py import MPI
+from brainiak.searchlight.searchlight import Searchlight
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "MVPAVoxelSelector",
+]
+
+
+class MVPAVoxelSelector:
+    """Activity-based voxel selection component of FCMA
+
+    Parameters
+    ----------
+
+    raw\_data: list of 4D array
+        Assumption: 1. all activity data contains the same number of voxels
+                    2. the activity data has been z-scored,
+                       ready to compute correlation as matrix multiplication
+                    3. all subjects have the same number of epochs
+                    4. epochs belonging to the same subject are adjacent
+                       in the list
+                    5. voxel selection is always done in the auto-correlation,
+                       i.e. raw_data correlate with themselves
+                    6. if MPI jobs are running on multiple nodes, the user
+                       home directory is shared by all nodes
+
+    mask: 3D array
+
+    epoch\_info: list of tuple (label, sid, start, end)
+        the condition labels of the epochs
+        len(labels) labels equals the number of epochs
+        assuming the epochs of the same sid are adjacent,
+        and sid is the index of raw\_data
+
+    num\_folds: int
+        the number of folds to be conducted in the cross validation
+
+    sl: Searchlight
+        the distributed Searchlight object
+
+    processed\_data: 4D array in shape [brain 3D + epoch]
```

Comment:
The parameter list here documents the properties of `MVPAVoxelSelector` instead of the input of `__init__`. The two voxel selector of FCMA is not about fitting, so I didn't follow the standard to only pass data through `fit`. I agree to keep the code structure as it is and leave the restructure discussion later. For now, let me modify the properties not passed through `__init__` as `var_`.